docs/mps.md

```diff
@@ -0,0 +1,91 @@
+
+First, we measure the throughput of a single worker for different batch sizes as it will show us at which point the compute resources of the GPU are fully occupied.
+Second, we measure the throughput with two deployed workers for the batch sizes where we expect the GPUs to have still some resources left over to share.
+For each benchmark we perform five runs and take the median over the runs.
+
+We use the following config.json for the benchmark, only overwriting the number of workers and the batch size accordingly.
+
+```
+{
+    "url":"/home/ubuntu/serve/examples/Huggingface_Transformers/model_store/BERTSeqClassification",
+    "requests": 10000,
+    "concurrency": 600,
+    "input": "/home/ubuntu/serve/examples/Huggingface_Transformers/Seq_classification_artifacts/sample_text_captum_input.txt",
+    "workers": "1"
+}
+```
+Please note that we set the concurrency level to 600 which will make sure that the batch aggregation inside TorchServe fills up the batches to the maximum batch size. But concurrently this will skew the latency measurements as many requests will be waiting in the queue to be processed. We will therefore neglect the latency measurements in the following.
+
+### G4 Instance
+We first perform the single worker benchmark for the G4 instance.
+In the figure below we see that up to a batch size of four we see a steady increase of the throughput over the batch size.
+
+![G4 benchmark, single worker](images/mps_g4_single.png)
+
+Next, we increase the number of workers to two in order to compare the throughput with and without MPS running.
+To enable MPS for the second set of runs we first set the exclusive processing mode for the GPU and then start the MPS daemon as shown above.
+
+We select the batch size between one and eight according to our previous findings.
+In the figure we can see that the performance in terms of throughput can be better in case of batch size 1 and 8 (up to +18%) while it can be worse for others (-11%).
+An interpretation of this result could be that the G4 instance has not many resources to share when we run a BERT model in one of the workers.
+
+![G4 benchmark, two workers](images/mps_g4_two_worker.png)
+
+### P3 instance
+Next, we will run the same experiment with the bigger p3.2xlarge instance.
+With a single worker we get the following throughput values:
+
+![P3 benchmark, single worker](images/mps_p3_single.png)
+
+We can see that the throughput steady increases but for a batch size over eight we see diminishing returns.
+Finally, we deploy two workers on the P3 instance and compare running them with and without MPS.
+We can see that for batch size between 1 and 32 the throughput is consistently higher (up to +25%) for MPS enabled with the exception of batch size 16.
+
+![P3 benchmark, two workers](images/mps_p3_two_worker.png)
+
+## Summary
```

Comment:
can we add some information like
once gpu utilization hits threshold (eg. 50%+), the throughput gain is limited.